File: app/src/main/java/com/example/fooddeuk/adapter/RestaurantAdapter.java
package com.example.fooddeuk.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.fooddeuk.R;
import com.example.fooddeuk.model.restaurant.Restaurant;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by heojuyeong on 2017. 10. 9..
 */

public class RestaurantAdapter extends RecyclerView.Adapter<RestaurantAdapter.ViewHolder> {
    private Context context;
    public ArrayList<Restaurant> restaurantItem;
    private RestaurantItemClickListener restaurantItemClickListener;
//    HashMap<String, Integer> starDpMap;


    public RestaurantAdapter(Context context, ArrayList<Restaurant> restaurantItem) {
        this.context = context;
        this.restaurantItem = restaurantItem;
//        starDpMap = LayoutUtil.DpToLayoutParams(context, 12, (float) 11.5);
    }



    @Override
    public RestaurantAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_restaurant, parent, false);
        // set the view's size, margins, paddings and layout parameters
        RestaurantAdapter.ViewHolder vh = new RestaurantAdapter.ViewHolder(v);
        return vh;
    }


    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Restaurant item = restaurantItem.get(position);

        Picasso.with(context).load(item.picture).resize(150, 150).into(holder.restaurantImageInList);
        holder.restaurantNameInList.setText(item.name);
        holder.restaurantReviewCountInList.setText(String.valueOf(item.reviewCnt));
        holder.restaurantAdminCommentInList.setText("todo");
        holder.restaurantRatingStarInList.removeAllViews();
//        setStarView(item.rating, holder.restaurantRatingStarInList);


    }

    @Override
    public int getItemCount() {
        return restaurantItem.size();
    }




    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        @Override
        public void onClick(View v) {
            restaurantItemClickListener.onItemClickListener(restaurantItem.get(getAdapterPosition()));
        }

        @BindView(R.id.restaurantImageInList)
        ImageView restaurantImageInList;
        @BindView(R.id.restaurantNameInList)
        TextView restaurantNameInList;
        @BindView(R.id.restaurantReviewCountInList)
        TextView restaurantReviewCountInList;
        @BindView(R.id.restaurantAdminCommentInList)
        TextView restaurantAdminCommentInList;
        @BindView(R.id.restaurantRatingStarInList)
        LinearLayout restaurantRatingStarInList;

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
            itemView.setOnClickListener(this);
        }
    }

//    private void setStarView(float rating, LinearLayout restaurantRatingStarInList) {
//        for (int j = 0; j < 5; j++) {
//            ImageView star = new ImageView(context);
//            star.setLayoutParams(new ViewGroup.LayoutParams(starDpMap.get("width"), starDpMap.get("height")));
//            if (rating == 0) {
//                star.setImageResource(R.drawable.ic_star);
//                restaurantRatingStarInList.addView(star);
//
//            } else if (rating >= 1.0) {
//                rating -= 1.0;
//                star.setImageResource(R.drawable.ic_star_ranked);
//                restaurantRatingStarInList.addView(star);
//
//            } else if (rating == 0.5) {
//                rating -= 0.5;
//                star.setImageResource(R.drawable.ic_star_ranked_half);
//                restaurantRatingStarInList.addView(star);
//
//            }
//        }
//    }

    public interface RestaurantItemClickListener {
        void onItemClickListener(Restaurant restaurant);
    }

    public void setRestaurantItemClickListener(RestaurantItemClickListener restaurantItemClickListener) {
        this.restaurantItemClickListener = restaurantItemClickListener;
    }

    public void updateRestaurant(ArrayList<Restaurant> restaurant){
            restaurantItem.clear();
            restaurantItem.addAll(restaurant);
            notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/fooddeuk/adapter/RestaurantAdapter.kt
package com.example.fooddeuk.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.fooddeuk.R
import com.example.fooddeuk.model.restaurant.Restaurant
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.item_restaurant.view.*
import java.util.*

/**
 * Created by heojuyeong on 2017. 10. 9..
 */

class RestaurantAdapter
//    HashMap<String, Integer> starDpMap;


(private val context: Context, var restaurantItem: ArrayList<Restaurant>)//        starDpMap = LayoutUtil.DpToLayoutParams(context, 12, (float) 11.5);
    : RecyclerView.Adapter<RestaurantAdapter.ViewHolder>() {
    private var restaurantItemClickListener: RestaurantItemClickListener? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RestaurantAdapter.ViewHolder {

        // set the view's size, margins, paddings and layout parameters
        val viewHolder = ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_restaurant, parent, false))
        viewHolder.itemView.setOnClickListener({restaurantItemClickListener?.onItemClickListener(restaurantItem[viewHolder.adapterPosition])})
        return viewHolder
    }


    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = restaurantItem[position]
        holder.bind(item)

        //        holder.star_restaurant_list.setNumStars(item.rating);

        //        setStarView(item.rating, holder.restaurantRatingStarInList);


    }

    override fun getItemCount(): Int = restaurantItem.size


    inner class ViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView){

        fun bind(restaurant: Restaurant){
            with(itemView){
                Picasso.with(context).load(restaurant.picture).fit().into(img_restaurant_list)
                txt_restaurant_name_list.text=restaurant.name
                star_restaurant_list.rating=restaurant.rating
                txt_restaurant_reviewCnt_list.text="리뷰(${restaurant.reviewCnt})"

                if(restaurant.discount.isNotEmpty()) {
                    txt_restaurant_sale_list.text="포장 시 최대 -${restaurant.discount}"
                } else txt_restaurant_sale_list.visibility=View.GONE
            }

        }

    }



    interface RestaurantItemClickListener {
        fun onItemClickListener(restaurant: Restaurant)
    }

    fun setRestaurantItemClickListener(restaurantItemClickListener: RestaurantItemClickListener) {
        this.restaurantItemClickListener = restaurantItemClickListener
    }

    fun updateRestaurant(restaurant: ArrayList<Restaurant>) {
        restaurantItem.clear()
        restaurantItem.addAll(restaurant)
        notifyDataSetChanged()
    }
}
